### PR TITLE
Feature/942 add phone no searchrequest

### PR DESCRIPTION
**Pull request by rantifamilusi**
# Description

Add options to check for lists for addresses and phones. Refactored to be generic. There is still opportunity to refactor.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Locally unit tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **FAMS3-928 & FAMS3-946 **

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs [hunk 1]
```diff
@@ -54,19 +54,47 @@ private async Task<bool> CheckStatusReason(CancellationToken cancellationToken)
...
-            foreach (var identificationType in Enumeration.GetAll<IdentificationType>())
+            foreach (var optionType in optionTypes)
             {
-                if (!idTypes.Any(x => x.Value == identificationType.Value && string.Equals(x.Name, identificationType.Name, StringComparison.OrdinalIgnoreCase)))
-                    return false;
+                var types = await _optionSetService.GetAllOptions(optionType, cancellationToken);
+                foreach (var entityType in GetListOfOptions(optionType))
+                {
+                    if (!types.Any(x => x.Value == entityType.Value && string.Equals(x.Name, entityType.Name, StringComparison.OrdinalIgnoreCase)))
+                        return false;
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: can we add some log here, then from log, we can know which optionSet is mismatched.

File: app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 using System.Threading;
 using System.Threading.Tasks;
 using DynamicsAdapter.Web.SearchRequest;
-using Fams3Adapter.Dynamics.Identifier;
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: did u forget to change the filename?

Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: What file ?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs (hunk 1 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: good catch. log added

**Pull request merged**